File: leetcode/06-Valid-Parentheses/python/main.py
class Solution:
    def isValid(self, s: str) -> bool:
        result = True
        stack = []

        for currentChar in s:
            if currentChar == "(" or currentChar == "[" or currentChar == "{":
                stack.append(currentChar)
            elif currentChar == ")" and len(stack) > 0 and stack[-1] == "(":
                stack.pop()
            elif currentChar == "]" and len(stack) > 0 and stack[-1] == "[":
                stack.pop()
            elif currentChar == "}" and len(stack) > 0 and stack[-1] == "{":
                stack.pop()
            else:
                result = False
                break

        if len(stack) > 0:
            result = False

        return result

File: leetcode/06-Valid-Parentheses/go/main.go
package main

func isValid(s string) bool {
	var stack []rune
	result := true

	for _, char := range s {
		if char == '(' || char == '[' || char == '{' {
			stack = append(stack, char)
		} else if len(stack) > 0 && stack[len(stack)-1] == '(' && char == ')' {
			stack = stack[:len(stack)-1]
		} else if len(stack) > 0 && stack[len(stack)-1] == '{' && char == '}' {
			stack = stack[:len(stack)-1]
		} else if len(stack) > 0 && stack[len(stack)-1] == '[' && char == ']' {
			stack = stack[:len(stack)-1]
		} else {
			result = false
			break
		}
	}

	if len(stack) > 0 {
		result = false
	}
	return result
}
